src/map_transition.js

```diff
@@ -156,42 +152,34 @@ function mapTransition (viewId, map, fitBoundsOptions) {
   }
 }
 
-function setLayerOpacity (map, layerId, opacity, duration) {
+function setLayerOpacity (map, layerId, visible, duration) {
   if (typeof duration === 'undefined') duration = FADEIN_DURATION
   var layer = map.getLayer(layerId)
-  var propNames = getOpacityPropNames(layer)
-  // Workaround for https://github.com/mapbox/mapbox-gl-js/issues/6706
-  // Need to call `setPaintProperty()` on the layer, not `map`
-  propNames.forEach(function (propName) {
-    // TODO: are these always guaranteed to be in the map?
-    // layer.setPaintProperty(propName + '-transition', {duration: duration, delay: 0})
-    // map.setPaintProperty(layerId, propName, opacity)
+  var props = getOpacityProps(layer)
+  props.forEach(function ({name, opacity}) {
+    map.setPaintProperty(layerId, name, visible === 0 ? 0 : opacity)
   })
 }
 
-function expandLayerGlobs (map) {
-  var mapLayers = map.getStyle().layers.map(function (l) {
-    return l.id
-  })
-  hiddenLayersExpanded = mm(mapLayers, HIDDEN_TRANSITION_LAYERS)
-  Object.keys(views).forEach(function (key) {
-    var expandedLayerOpacity = {}
-    var layerMatchPatterns = Object.keys(views[key].layerOpacity)
-    layerMatchPatterns.forEach(function (pattern) {
-      var matchedLayers = mm(mapLayers, [pattern])
-      matchedLayers.forEach(function (layerId) {
-        expandedLayerOpacity[layerId] = views[key].layerOpacity[pattern]
-      })
+function getOpacityProps (layer) {
```

Comment:
get opacity from the map. something broken here, perhaps I do not understand how these layers/paint properties interact well enough. need to read up!